java-api-core/src/com/kinvey/java/store/BaseDataStore.java

```diff
@@ -417,6 +434,17 @@ public void purge() {
         client.getSyncManager().clear(collection);
     }
 
+    public void purge(Query query) {
+        Preconditions.checkArgument(storeType != StoreType.NETWORK, "InvalidDataStoreType");
+        Preconditions.checkNotNull(client, "client must not be null.");
+        Preconditions.checkArgument(client.isInitialize(), "client must be initialized.");
+        for (T item : cache.get(query)) {
+            if (item.get("_id") != null) {
```

Comment:
Since item.get("_id") is potentially being called twice here, we should save the value to a local variable and use it twice.

Reply:
Improved